### PR TITLE
Add shelley cardano-cli error rendering functions

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Run.hs [hunk 1]
```diff
@@ -78,12 +92,22 @@ runGovernanceCmd (GovernanceUpdateProposal out eNo genVKeys ppUp) =
 runGovernanceCmd cmd = liftIO $ putStrLn $ "TODO: runGovernanceCmd: " ++ show cmd
 
 data ShelleyGovernanceError
-  = CardanoApiError !ApiError
-  | EmptyUpdateProposal
-  | ReadKeyError !KeyError
-
+  = GovernanceWriteUpateError !FilePath !ApiError
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run.hs:95]:
```suggestion
  = GovernanceWriteUpdateError !FilePath !ApiError
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs [hunk 2]
```diff
@@ -49,16 +51,65 @@ import           Cardano.Config.Shelley.ColdKeys
...
+    ShelleyGenesisCmdUTxOKeyGenError keyErr ->
+      "Error while generating the genesis UTxO keys:" <> renderShelleyKeyGenError keyErr
+    ShelleyGenesisCmdReadVerKeyError keyErr ->
+      "Error while reading genesis verification key: " <> renderKeyError keyErr
+    ShelleyGenesisCmdReadSignKeyError keyErr ->
+      "Error while reading the genesis signing key: " <> renderKeyError keyErr
+    ShelleyGenesisCmdReadGenesisUTxOVerKeyError keyErr ->
+      "Error while reading the genesis UTxO verification key: " <> renderKeyError keyErr
+    ShelleyGenesisCmdOnlyShelleyAddressesNotReward ->
+      "Please only supply shelley addresses. Reward accound found."
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs:87]:
```suggestion
      "Please only supply Shelley addresses. Reward account found."
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs [hunk 3]
```diff
@@ -49,16 +51,65 @@ import           Cardano.Config.Shelley.ColdKeys
...
+  | ShelleyGenesisCmdOnlyShelleyAddressesNotReward
+  | ShelleyGenesisCmdOperatorKeyGenError !ShelleyKeyGenError
+  | ShelleyGenesisCmdReadGenesisAesonDecodeError !FilePath !Text
+  | ShelleyGenesisCmdReadGenesisIOError !FilePath !IOException
+  | ShelleyGenesisCmdReadGenesisUTxOVerKeyError !KeyError
+  | ShelleyGenesisCmdReadIndexedVerKeyError !KeyError
+  | ShelleyGenesisCmdReadSignKeyError !KeyError
+  | ShelleyGenesisCmdReadVerKeyError !KeyError
+  | ShelleyGenesisCmdUTxOKeyGenError !ShelleyKeyGenError
+  | ShelleyGenesisCmdWriteDefaultGenIOError !FilePath !IOException
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs:70]: For consistency with the other error constructors, I would use "genesis' here:

```suggestion
  | ShelleyGenesisCmdWriteDefaultGenesisIOError !FilePath !IOException
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Node.hs [hunk 4]
```diff
@@ -14,23 +15,69 @@ import           Cardano.Config.Shelley.OCert
...
+  | ShelleyNodeWriteKESSignKeyError !FilePath !KESError
+  | ShelleyNodeWriteVRFSignKeyError !FilePath !VRFError
+  | ShelleyNodeWriteVRFVerKeyError !FilePath !VRFError
+  | ShelleyNodeReadOperationalCertCounterError !FilePath !OperationalCertError
+  | ShelleyNodeOperatorKeyGenError
+      !FilePath
+      -- ^ Verification key
+      !FilePath
+      -- ^ Signing key
+      !ShelleyKeyGenError
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Node.hs:40]: Doesn't need to change, I just wonder whether this is more clear than using two `FilePath`s:

```suggestion
  | ShelleyNodeOperatorKeyGenError
      !VerificationKeyFile
      !SigningKeyFile
      !ShelleyKeyGenError
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs [hunk 5]
```diff
@@ -52,9 +53,22 @@ import           Shelley.Spec.Ledger.Keys
 data ShelleyQueryCmdError
   = ShelleyQueryEnvVarSocketErr !EnvSocketError
   | NodeLocalStateQueryError !LocalStateQueryError
-  | ShelleyHelperError !HelpersError
+  | ShelleyQueryWriteProtocolParamsError !FilePath !IOException
+  | ShelleyQueryWriteFilteredUTxOsError !FilePath !IOException
+  | ShelleyQueryWriteStakeDistribError !FilePath !IOException
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs:58]: Not necessary, but I would use the full word, "distribution":

```suggestion
  | ShelleyQueryWriteStakeDistributionError !FilePath !IOException
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs [hunk 6]
```diff
@@ -52,9 +53,22 @@ import           Shelley.Spec.Ledger.Keys
...
+renderShelleyQueryCmdError err =
+  case err of
+    ShelleyQueryEnvVarSocketErr envSockErr -> renderEnvSocketError envSockErr
+    NodeLocalStateQueryError lsqErr -> renderLocalStateQueryError lsqErr
+    ShelleyQueryWriteProtocolParamsError fp ioException ->
+      "Error writing protocol parameters at: " <> textShow fp <> " Error: " <> textShow ioException
+    ShelleyQueryWriteFilteredUTxOsError fp ioException ->
+      "Error writing filtered UTxOs at: " <> textShow fp <> " Error: " <> textShow ioException
+    ShelleyQueryWriteStakeDistribError fp ioException ->
+      "Error writing at: " <> textShow fp <> " Error: " <> textShow ioException
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs:71]:
```suggestion
      "Error writing stake distribution at: " <> textShow fp <> " Error: " <> textShow ioException
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs [hunk 7]
```diff
@@ -32,15 +35,40 @@ import           Shelley.Spec.Ledger.PParams (PParams)
...
 data ShelleyTxCmdError
-  = ShelleyTxSocketEnvError !EnvSocketError
-  | ShelleyTxCardanoApiError !ApiError
-  | ShelleyTxCardanoIOError !FilePath !IOException
-  | ShelleyTxCardanoAesonDecode !FilePath !Text
-  | ShelleyTxCertReadError !ApiError
-  | ShelleyTxTxKeyError !KeyError
+  = ShelleyTxAesonDecodeProtocolParamsError !FilePath !Text
+  | ShelleyTxSocketEnvError !EnvSocketError
+  | ShelleyTxReadProtocolParams !FilePath !IOException
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:40]: Also not necessary to change, but:

```suggestion
  | ShelleyTxReadProtocolParamsError !FilePath !IOException
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Node.hs (hunk 4 above)
Comment (erikd) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Node.hs:40]: :100: I think that is a change well worth making.

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs [hunk 8]
```diff
@@ -76,16 +127,16 @@ runGenesisCmd (GenesisCreate gd gn un ms am) = runGenesisCreate gd gn un ms am
...
 
 runGenesisKeyGenDelegate :: VerificationKeyFile
                          -> SigningKeyFile
                          -> OpCertCounterFile
                          -> ExceptT ShelleyGenesisCmdError IO ()
 runGenesisKeyGenDelegate vkeyPath skeyPath (OpCertCounterFile ocertCtrPath) = do
-    firstExceptT ShelleyGenesisCmdKeyGenError $ runColdKeyGen (OperatorKey GenesisDelegateKey) vkeyPath skeyPath
-    firstExceptT ShelleyGenesisCmdOperationalCert $
+    firstExceptT ShelleyGenesisCmdOperatorKeyGenError $ runColdKeyGen (OperatorKey GenesisDelegateKey) vkeyPath skeyPath
+    firstExceptT (ShelleyGenesisCmdWriteOperationalCertError ocertCtrPath) $
```
Comment (erikd) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs:139]: :100:

**Pull request merged**